### DISCUSSION TITLE
URL Data Type

### DISCUSSION THREAD

**Discussion by gramian** [ArcadeData/arcadedb#609] [Ideas]
Dear Developers,

I would like to suggest a `URL` data type.

Beyond websites, `URL`s (or [`URI`](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier)s) have widespread application as identifiers, for example in academia, digital objects (publications, datasets, software) are identified by a [DOI](https://en.wikipedia.org/wiki/Digital_object_identifier), while scientists are identified by an [ORCID](https://en.wikipedia.org/wiki/ORCID); for further URI schemes see https://en.wikipedia.org/wiki/List_of_URI_schemes .
Furthermore, knowledge graphs (in the [RDF](https://en.wikipedia.org/wiki/Resource_Description_Framework) model) are represented by triples of URIs.

Such a type could be derived from Java's [`URL` class](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/URL.html), but I don't know if it would fit ArcadeDB's low-level Java principle. However, it would prevent selecting a [suitable RegExp](https://mathiasbynens.be/demo/url-regex) for validating a string.

Thank you for considering

**Comment by lvca** [comment]
As you mentioned, you can store it in a string and if you need a constraint, use regexp. Having a URL type wouldn't add much to ArcadeDB IMO.
